pkgs/build-support/bintools-wrapper/default.nix

```diff
@@ -87,7 +87,7 @@ let
     else if targetPlatform.isDarwin                   then "/usr/lib/dyld"
     else if targetPlatform.isFreeBSD                  then "/libexec/ld-elf.so.1"
     else if lib.hasSuffix "pc-gnu" targetPlatform.config then "ld.so.1"
-    else null;
+    else "";
```

Comment:
This is not a general pattern, is it? Having more to do with good bash style, perhaps setting `set -u` at some point? (Treat unset as error)

It would be very useful to have a small section in the manual about upgrading a derivation to structured attrs.